src/sentry/models/commitfilechange.py

```diff
@@ -40,3 +41,18 @@ class Meta:
     @staticmethod
     def is_valid_type(value: str) -> bool:
         return value in COMMIT_FILE_CHANGE_TYPES
+
+
+def process_resource_change(instance, **kwargs):
+    from sentry.tasks.codeowners import code_owners_auto_sync
+
+    # CODEOWNERS file added or modified, trigger auto-sync
+    if "codeowners" in instance.filename.lower() and instance.type in ["A", "M"]:
```

Comment:
Would `src/sentry/api/endpoints/project_codeowners.py` get caught by this?

Reply:
hmm good catch! yea it would. I saw on the GitHubIntegration class we hardcode the paths and we can just search for those. I was thinking about the case for multi-package monorepos (multiple CODEOWNERS relative to each package). We can always refactor for this use-case when surfaced by users.